doc/source/whatsnew/v0.17.0.txt

```diff
@@ -330,6 +333,47 @@ Other API Changes
   ``raise ValueError``                All other public methods (names not beginning with underscores)
   ===============================     ===============================================================
 
+
+- default behavior for HDF write functions with "table" format is now to keep rows that are all missing except for index. (:issue:`9382`)
+
+Previously,
+
+.. code-block:: python
+
+   In [1]: 
+   df_with_missing = pd.DataFrame({'col1':[0, np.nan, 2], 'col2':[1, np.nan, 3]})
+   df_with_missing.to_hdf('file.h5', 'df_with_missing', format = 't')
+   
+   pd.read_hdf('file.h5', 'df_with_missing')
+   
+   Out [1]: 
+         col1  col2
+     0     0     1
+     2     2     3
+
+
```

Comment:
add a similar example (but just the new behavior of course) to a section in ``io.rst`` (and put a warning box that the default changed in 0.17.0